### DISCUSSION TITLE
How much vocabulary range it has?

### DISCUSSION THREAD

**Discussion by jaytimbadia** [mrmaxguns/wonderwordsmodule#11] [General]
Hey there, 
I am lucky I got here, exactly what I was in need of.

Just need to ask, which dictionary are you using to get the random word?
Also is there any method to incorporate words from fantasy novels and other genres?

If yes, will methods like starts_with and ends_with work for newly added words?

Thanks in advance!!

**Comment by mrmaxguns** [comment]
Hello! I'm glad you found my library useful.

Currently, I am using 4 dictionaries for nouns, verbs, adjectives and curse words (curse words are disabled by default). They're found [here](https://github.com/mrmaxguns/wonderwordsmodule/tree/master/wonderwords/assets).

With Wonderwords version 2.2 (the latest version; upgrade with `pip install --upgrade wonderwords`), there's built-in functionality to include your own custom word lists. Assuming you have `fantasy_words.txt` in the current folder, you could do something like this:

```python
from wonderwords import RandomWord

# Read the text file with the words from fantasy novels
# and add the words to a list.
with open("fantasy_words.txt") as f:
    words = f.readlines()

# Now remove newlines (\n) from the words
words = [i.strip() for i in words]
# Finally, remove empty strings (caused by extra
# newlines at the end of the file).
words = [i for i in words if i]

# Here's where the magic happens. We create a RandomWord
# object with our list of words. Here, we create a category called
# 'fantasy_words' and set it to our list of words. This category
# is like nouns and verbs, except for our own custom words.
generator = RandomWord(fantasy_words=words)

# Finally, we can perform all the same operations with our
# custom words:
generator.word(starts_with="dr") # dragon
```

The example above is a brief summary. If you want to learn more about custom categories, check out the [advanced tutorial](https://wonderwords.readthedocs.io/en/latest/advanced.html).

I hope that helps! If you have any more questions or problems, please let me know.

**Comment by mrmaxguns** [comment]
Since this is a question about the use-case, I'll convert it to a discussion.

**Reply by mrmaxguns** [reply]
Done.

**Reply by jaytimbadia** [reply]
Really appreciate your time. 

Thank you for the solution!!
